addons/mail/models/mail_message.py

```diff
@@ -179,6 +179,7 @@ def default_get(self, fields):
     mail_ids = fields.One2many('mail.mail', 'mail_message_id', string='Mails', groups="base.group_system")
     canned_response_ids = fields.One2many('mail.shortcode', 'message_ids', string="Canned Responses", store=False)
     reaction_ids = fields.One2many('mail.message.reaction', 'message_id', string="Reactions", groups="base.group_system")
+    source_lang = fields.Char('Source Language', help='Result of the language detection based on its content.')
```

Comment:
It might be better to keep `source_lang` on the translation table as well to keep message as light as possible.
It's a small duplication that has less impact than having a field here.

Reply:
If you consider data duplication to be more convenient than polluting the "mail.message" model, I will comply.